galley/pkg/config/analysis/analyzers/namespaceconflict/namespaceselector.go

```diff
@@ -0,0 +1,235 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespaceconflict
+
+import (
+	"fmt"
+
+	k8s_labels "k8s.io/apimachinery/pkg/labels"
+
+	v1beta1 "istio.io/api/security/v1beta1"
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pilot/pkg/util/sets"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+)
+
+// Analyzer checks conditions related to conflicting namespace level resources
+// Validates that there's only a single rule associated with a resource.
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{}
+
+var (
+	peerAuthCollection    = collections.IstioSecurityV1Beta1Peerauthentications.Name()
+	requestAuthCollection = collections.IstioSecurityV1Beta1Requestauthentications.Name()
+)
+
+func (a *Analyzer) Metadata() analysis.Metadata {
+	return analysis.Metadata{
+		Name:        "namespaceconflict.Analyzer",
+		Description: "Checks conditions related to conflicting namespace level resources",
+		Inputs: collection.Names{
+			peerAuthCollection,
+			requestAuthCollection,
+		},
+	}
+}
+
+func (a *Analyzer) Analyze(c analysis.Context) {
+	// Analyze collections.IstioSecurityV1Beta1Peerauthentications
+	c.ForEach(peerAuthCollection, func(r *resource.Instance) bool {
+		x := r.Message.(*v1beta1.PeerAuthentication)
+		if x.GetSelector() != nil {
+			// If the resource has workloads associated with it, analyze for conflicts with selector
+			a.analyzeWorkloadSelectorConflicts(r, c)
+		} else {
+			// If the resource has workloads associated with it, analyze for conflicts with namespace
+			a.analyzeNamespaceWideConflicts(r, c)
+		}
+		return true
+	})
+
+	// Analyze collections.IstioSecurityV1Beta1Requestauthentications
+	c.ForEach(requestAuthCollection, func(r *resource.Instance) bool {
```

Comment:
Actually multiple request authn for same workload is allowed, we just merge them so that all request authn policies will have effect at the same time.

Unlike the peer authN, there is no conflict here as it is just to accept multiple issuers configured at the same time.